### ISSUE TITLE
[F.Rq.] Gravatar

### ISSUE THREAD

**Issue by YoshiRulz** [TASVideos/tasvideos#640]
tl;dr: allows users to change their avatar on many sites at once, and auto-generates unique avatars when a user doesn't provide one. You've probably seen the latter e.g. on GitHub. [API docs](https://en.gravatar.com/site/implement)

update 2024-12: TIL there's an open alternative to Gravatar called [ivatar](https://wiki.libravatar.org/features/). It says it can fall back to Gravatar so maybe it's a direct upgrade?

see also #227

**Comment by meshuggahtas** [comment]
I'm not sure what does the first half means (change their avatar on many sites at once).

Personally I hate auto generated avatars, as it looks like I have picked specifically an idiot avatar I never choose. Just look at my auto generated green cactus gravatar.

Personally I hope this comes with the ability to not use any profile at all.

Since we don't really have a designated designers, I'm not sure about how much effort is required to have a fine selection of usable avatars.

**Comment by ActuallyTruncated** [comment]
I disagree with the closing of #227 in favor of this solution. In fact, I do not think this suggestion is a solution at all - it is more like deciding to keep having this problem, only choosing that it will occur for everyone simultaneously when Gravatar stops serving free bandwidth.

If you think to yourself "nah, that will never happen", consider that we once had ICQ and MSN Messenger integrations.

**Comment by adelikat** [comment]
going with your example, while ICQ and MSN might not be a good idea for integration today, it doesn't mean that going back in time, it would have been a good idea to have a custom messaging system as an alternative.  By the time those things aren't relevant, so was the feature.

Gravatar might not exist, but the concept seems like it can reasonably exist as it solves a current problem for at least as long as the avatar problem stays relevant.  Assuming this is a simple integration (still looking at the docs), should be easy enough to do something else should the day come.

**Comment by YoshiRulz** [comment]
Here's an idea for a UI to handle multiple sources, taken from StackOverflow (in case you add OAuth for login with Discord, or autogenerated "identicons", etc.):
![Screenshot_20220922_005241](https://user-images.githubusercontent.com/13409956/191538281-ee7c077b-85f9-4147-a7c6-fca977957cd3.png)

**Comment by adelikat** [comment]
Can we please make a separate request for some of the improvements.  I have no issue with identicons, probably, but I would like to let this old issue die, the original intent has been delivered
